### PR TITLE
[CORE-695] - add queries for x/perpetuals state

### PR DESCRIPTION
**Pull request by clemire**
### Changelist
Add gRPC and cli query commands to access perpetuals state.

### Test Plan
Added cli, qRPC tests.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

**Comment by ttl33** [comment]
@clemire can you link this PR to a different linear ticket and close out the "audit" ticket? I think we should track this work separately

**Comment by clemire** [comment]
https://linear.app/dydx/issue/CORE-695/xperpetuals-grpc-query-endpoints-for-state

**Comment by clemire** [comment]
> @clemire can you link this PR to a different linear ticket and close out the "audit" ticket? I think we should track this work separately

Yes, looks like I swapped two digits in the ticket number in the title, sorry about that. This work is unrelated.

**Comment by teddyding** [comment]
Could you fill out PR template?

File: protocol/x/perpetuals/client/cli/query_params.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"context"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	"github.com/spf13/cobra"
+)
+
+func CmdQueryParams() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-params",
```
Comment (teddyding) [inline protocol/x/perpetuals/client/cli/query_params.go:13]: Nit: it looks like this is conventionally just `params` in Cosmos SDK ([example](https://github.com/dydxprotocol/cosmos-sdk/blob/b95c66dedbd525a1b3d0f6b851d4fd351887fa23/x/slashing/client/cli/query.go#L114))

So the cmd looks like`dydxprotocold query perptuals params`

Up to you on whether to change the other queries, but for params probably makes sense to stick to conventions

File: protocol/x/perpetuals/client/cli/query_premiums.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdQueryPremiumVotes() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-premium-votes",
+		Short: "get the PremiumVotes",
```
Comment (teddyding) [inline protocol/x/perpetuals/client/cli/query_premiums.go:37]: Could you update this to "Get PremiumVotes from the current `funding-sample` epoch"

File: protocol/x/perpetuals/client/cli/query_premiums.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	"github.com/spf13/cobra"
+)
+
+func CmdQueryPremiumSamples() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-premium-samples",
+		Short: "get the PremiumSamples",
```
Comment (teddyding) [inline protocol/x/perpetuals/client/cli/query_premiums.go:14]: Could you update this to "Get PremiumSamples from the current `funding-tick` epoch"

**Review by teddyding — APPROVED** [review]

File: protocol/x/perpetuals/client/cli/query_params.go (hunk 1 above)
Comment (clemire) [reply protocol/x/perpetuals/client/cli/query_params.go:13]: I actually copied the style in x/bridge - I think I might keep it this way but file a follow-up ticket to migrate to cosmos style.

Comment (clemire) [reply protocol/x/perpetuals/client/cli/query_params.go:13]: https://linear.app/dydx/issue/CORE-716/change-grpc-getparams-queries-to-remove-get-from-cli-command

**Pull request merged**